packages/next/src/bin/next.ts

```diff
@@ -40,6 +41,28 @@ if (
   process.exit(1)
 }
 
+// Native TypeScript resolution is supported with a flag
+// since v22.7.0, and is enabled by default since v23.6.0.
+// TODO: Remove this once we bump minimum Node.js version to v22
+if (
+  !process.execArgv.includes('--experimental-transform-types') &&
+  !process.env.NODE_OPTIONS?.includes('--experimental-transform-types') &&
+  semver.gte(process.versions.node, '22.7.0') &&
+  semver.lt(process.versions.node, '23.6.0')
+) {
+  const result = spawnSync(
+    process.argv[0],
+    [
+      '--experimental-transform-types',
+      '--disable-warning=ExperimentalWarning',
+      ...process.execArgv,
+      ...process.argv.slice(1),
+    ],
+    { stdio: 'inherit' }
+  )
```

Comment:
So we're just spawning another child e.g. `next dev`? That will break the debugger ports.

I was thinking we just don't do our routine if we detect [`process.features.typescript`](https://nodejs.org/api/process.html#processfeaturestypescript) instead.

That property won't exist in versions without TS stripping support. 

We should probably warn if we detect `--no-experimental-strip-types` when you have a `next.config.ts`. And that warning can be disabled in prod once we start loading resolved configs only in prod.